tests/Unit/Domain/Test_Domain.cpp

```diff
@@ -464,6 +475,592 @@ void test_3d_rectilinear_domains() {
   }
 }
 
+// The version testing requires an old version of the TimeDependentMapOptions
```

Comment:
Let's put all this into an anonymous namespace so we don't have to worry about it conflicting with anything. No need to keep it in the `bco` namespace